cmake/FindFILESYSTEM.cmake

```diff
@@ -0,0 +1,55 @@
+#  FILESYSTEM - filesystem component of the standard library
+#
+#  FILESYSTEM_FOUND       - system has std::filesystem
+#  FILESYSTEM_LIBRARIES   - libraries for std::filesystem
+#
+#  NOTE: it is not an official cmake search file
+#  authors: Carsten Uphoff, Ravil Dorozhinskii
+#  email: uphoff@in.tum.de, ravil.dorozhinskii@tum.de 
+#
+
+
+include(FindPackageHandleStandardArgs)
+include(CheckCXXSourceRuns)
+
+set(_FILESYSTEM_TEST_RPOGRAM "
+#include <filesystem>
+#include <iostream>
+
+namespace fs = std::filesystem;
+
+int main(int argc, char* argv[]) {
+    fs::path p{argv[0]};
+    std::cout << fs::canonical(p) << std::endl;
+    return 0;
+}")
+
+set(CMAKE_REQUIRED_DEFINITIONS -std=c++17)
```

Comment:
That's not so great as some compilers might enable c++17 with other flags.

Reply:
Hi, It turns out that there is no way how one can pass `CXX_STANDARD 17` for `check_cxx_source_runs` in a similar way how it is done for `try_compile` or `try_run`. The documentation says that `check_cxx_source_runs` internally calls `try_run`.

My solution is to save  `CXX_STANDARD` at the beginning, set it to 17 for the checks, and restore its original value at the end. 

Changes are pushed

Reply:
Btw. why not just use try_compile instead of cxx_source...

Reply:
Ahh, that was the problem. It turned out that one can compile the sample program with `gcc@8` without any issue. For example, I checked that `FILESYSTEM_LIBRARIES` in case of `gcc@8` was empty with the original cmake files. But I had a segmentation fault when I tried to run SeisSol. A better solution is to use `try_run` which does the job of `try_compile`, plus it tries to run a sample program.

`check_cxx_source_runs` internally calls `try_run`. Additionally, the macro handles interpreting output as `success` of `failure` as everybody expects (true or false), whereas,  `try_run` just gives you a raw output.
For example, if you write `return 0` then you will write `if (${RESULT} STREQUAL "0")`

Reply:
Ah ok

Reply:
The drawback is `check_cxx_source_runs` doesn't take `CXX_STANDARD 17` as a parameter in a similar way how `try_run` or `try_compile` does. I had to store the global value of `CXX_STANDARD` at the beginning, set it to `17` for the checks, and then restore this back to its original value at the end